base/client.jl

```diff
@@ -82,6 +82,12 @@ function ip_matches_func(ip, func::Symbol)
     return false
 end
 
+struct ExceptionInfo
```

Comment:
Is this the same as https://github.com/JuliaLang/julia/pull/29901/files#diff-113b7e6edf0ac665e7b9a45e50c50fa3292e68265b655fe6b33d5bef10ee05c8R126 ? @c42f

Reply:
It might be. The type of `lasterr` in the REPL loop that gets packed into this is `Vector{Tuple{Any, Vector{Union{Ptr{Nothing}, InterpreterIP}}}}`. I was thinking it might be better to pass around something more concrete to organize the information but that'd require a bit more reworking of the exception code to accommodate it. Also wasn't sure if sometimes you might have multiple errors, because it seems like it's usually just one.

Reply:
There can be multiple errors, if an exception is thrown while handling another one. Anyway if we're going to have this type, it seems like `catch_stack` should just return it in the first place instead of using it only in the REPL.

Reply:
If #29901 is going to be merged, then I'll wait to base this PR off of that.

Reply:
Yes, I think this is the same. The key reason for that struct was to have a nominal type for exception stacks (for which `show` can be overridden, etc) vs the current situation where exception stacks as returned from `catch_stack` can only be recognized structurally.

Reply:
Made this PR compatible with #29901